### ISSUE TITLE
Test of flashing yellow should accept dark on some signal groups

### ISSUE THREAD

**Issue by pksoft72** [rsmp-nordic/rsmp_validator#465]
This test switches to flashing yellow and expect all groups to have state 'c'
```ruby
    # Verify that we can yellow flash causes all groups to go to state 'c'
    #
    # 1. Given the site is connected
    # 2. Send the control command to switch to Yellow flash
    # 3. Wait for all groups to go to group 'c'
    # 4. Send command to switch to normal control
    # 5. Wait for all groups to switch do something else that 'c'
    specify 'yellow flash affects all signal groups', sxl: '>=1.0.7' do |example|
      Validator::Site.connected do |task,supervisor,site|
        prepare task, site
        timeout =  Validator.get_config('timeouts','yellow_flash')

        switch_yellow_flash
        wait_for_groups 'c', timeout: timeout      # c mean s yellow flash

        switch_normal_control
        wait_for_groups '[^c]', timeout: timeout   # not c, ie. not yellow flash
      end
    end

```
According to https://rsmp-nordic.github.io/rsmp_sxl_traffic_lights/1.2.1/signal_group_status.html, 'c' means "Manual control to flashing yellow" and output is "Yellow flash".

But there are signal groups - typically pedestrian or any single color groups - which are dark in yellow flash mode so I suggest to report 'a' = Disabled/dark for these groups and accept such groups in this test.

So I suggest change 1st wait_for_groups to

```ruby
wait_for_groups '([a,c]*c[a,c]*)', timeout: timeout      # c mean s yellow flash, a means dark
```

I could pass this test by removing all non R-Y-G signal groups, but I prefer real conditions.

**Comment by SwarcoPalm** [comment]
These are logical states, startup e,f,g does not tell you about the physical output, startup sequence is country dependent. This is how it is supposed to be

**Comment by otterdahl** [comment]
@pksoft72 Thank you for the feedback. I think you make a good point. However, as @SwarcoPalm points out, these are internal states and doesn't necessarily reflect the output. Also, we rely on the internal states in order to tell whether the command has been successful or not. If we were to accept that some signal groups goes to dark mode instead we also somehow need to specify the type of signal group (pedestrian, bicycle, vehicle) which would the configuration more complex and error prone. I'm not convinced that it is worth the trouble.

**Comment by pksoft72** [comment]
Ok, I have probably wrongly interpreted column Output in table
https://rsmp-nordic.github.io/rsmp_sxl_traffic_lights/1.2.1/signal_group_status.html
so when it says Red - I have expected, that it must be only Red and nothing else.

Startup sequence is country dependent, but if we hide real state, it is impossible to check through RSMP, whether startup sequence is ok on real intersection.

If these values are misinterpreted by Police, they could wrongly send some driver to jail.

**Comment by otterdahl** [comment]
> Startup sequence is country dependent, but if we hide real state, it is impossible to check through RSMP, whether startup sequence is ok on real intersection.

Correct. We cannot check this using RSMP - at least not completely. This is mostly true for the entire SXL. There is not way we can be sure that the actual content of SXL is interpreted correctly by the TLC or the supervision system. What we can verify, however, is that every part of the SXL is implemented (that we receive correct requests and responses) and RSMP works as intended. We do _some_ basic checks, such as the test case you mentioned, but it is no guarantee that is fulfills the legal requirements completely.

**Comment by otterdahl** [comment]
> @otterdahl Based on the above, my understanding is that the current tests should not be changed?

Correct. I think so.

**Comment by pksoft72** [comment]
I have changed behavior on Cross RS4S to report logical state.

**Comment by pksoft72** [comment]
So this issue is solved via reporting logical state.
